crates/test-utils/input/square.ll

```diff
@@ -0,0 +1,17 @@
+define dso_local i32 @square(i32) {
```

Comment:
This should not live in the utils crate. The main E2E tests should probably live in "driver", and so should the test inputs.

Also, where on earth did this come from? It needs a comment saying it. 

It is also really strange LLVM IR that is not really equivalent to the above cairo.

Reply:
I have generated the LLVM-IR code from the Rust compiler using a program equivalent to the Cairo code.

Reply:
This is not really equivalent. It is doing things that the cairo code does not with memory. I would suggest hand-writing it instead, and avoiding the pointers to just use SSA immediates instead. That would be more equivalent.